java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java

```diff
@@ -1704,6 +1706,12 @@ private static boolean updateSearchIndex() {
 
         // compute id to errata map
         List<Errata> errataList = ErrataManager.lookupErrataByIds(errataIds, user);
+        List<Errata> retracted = errataList.stream()
+                .filter(e -> e.getAdvisoryStatus() == AdvisoryStatus.RETRACTED)
+                .collect(toList());
+        if (!retracted.isEmpty()) {
+            throw new RetractedErrataException(retracted.stream().map(Errata::getId).collect(toList()));
```

Comment:
This introduces another cyclical dependency between java backend and xmlrpc. While the problem already exists in the codebase, I would advise not to "amplify" it further. The easiest thing to avoid this, is to declare 2 classes representing the exception on backend (`RetractedErrataException`) and xmlrpc (`RetractedErrataFault`) and use the `ExceptionTranslator` to convert one into another.